.github/workflows/image_build_push.yml

```diff
@@ -0,0 +1,144 @@
+        id: get_docker_tag
+        run: |
+          cat tag_file.txt
+          docker_image_tag=$(cat tag_file.txt)
+          echo $docker_image_tag
+          echo "docker_image_tag=$(echo $docker_image_tag)" >> $GITHUB_OUTPUT
+  build:
+    needs: fetch_tag
+
+    runs-on: ubuntu-latest
+
+    env:
+      DOCKER_IMAGE_TAG_1: ${{ needs.fetch_tag.outputs.docker_image_tag }}
+      DOCKER_IMAGE_TAG_2: ${{ github.event.inputs.docker_image_tag }}
+      
+    steps:
+    - uses: actions/checkout@v4
+
+    - name: Print input docker image tag
+      run: |
+        echo "Event name: ${{ github.event_name }}"
+        echo "Latest docker image tag (push): ${{ env.DOCKER_IMAGE_TAG_1 }}"
+        echo "Latest docker image tag (workflow_dispatch): ${{ env.DOCKER_IMAGE_TAG_2 }}"
+
+    - name: Update .env file 
+      run: |
+        if [ "${{ github.event_name }}" == "workflow_dispatch" ]; then
+          echo "Workflow_dispatch: New server image built and pushed, Updating image tag in .env"
+          echo "DOCKER_IMAGE_TAG=$DOCKER_IMAGE_TAG_2" > .env
+        else
+          echo "Push event: Restoring latest server image tag in .env"
+          echo "DOCKER_IMAGE_TAG=$DOCKER_IMAGE_TAG_1" > .env
+        fi
+
+    - name: Add, Commit, Push changes to .env file
+      run: |
+        git config --local user.email "action@github.com"
+        git config --local user.name "GitHub Action"
+        if git diff --quiet; then
+          echo "Latest timestamp already present in .env file, no changes to commit"
+        else
+          git add .env
+          git commit -m "Updated docker image tag in .env file to the latest timestamp"
+          git push origin 
+        fi
+
+    - name: docker login
+      run: | # log into docker hub account
+        docker login -u $DOCKER_USER -p $DOCKER_PASSWORD
+
+    - name: Get current date # get the date of the build
+      id: date
+      run: echo "::set-output name=date::$(date +'%Y-%m-%d--%M-%S')"
+
+    - name: Run a one-line script
+      run: echo running in repo ${GITHUB_REPOSITORY#*/} branch ${GITHUB_REF##*/} on ${{ steps.date.outputs.date }}
+
+    - name: build docker image
+      run: |
+        if [ "${{ github.event_name }}" == "workflow_dispatch" ]; then
+          docker build --build-arg DOCKER_IMAGE_TAG=$DOCKER_IMAGE_TAG_2 -t $DOCKER_USER/${GITHUB_REPOSITORY#*/}:${GITHUB_REF##*/}_${{ steps.date.outputs.date }} ./frontend
+          docker build --build-arg DOCKER_IMAGE_TAG=$DOCKER_IMAGE_TAG_2 -t $DOCKER_USER/${GITHUB_REPOSITORY#*/}_notebook:${GITHUB_REF##*/}_${{ steps.date.outputs.date }} ./viz_scripts
+        else
+          docker build --build-arg DOCKER_IMAGE_TAG=$DOCKER_IMAGE_TAG_1 -t $DOCKER_USER/${GITHUB_REPOSITORY#*/}:${GITHUB_REF##*/}_${{ steps.date.outputs.date }} ./frontend
+          docker build --build-arg DOCKER_IMAGE_TAG=$DOCKER_IMAGE_TAG_1 -t $DOCKER_USER/${GITHUB_REPOSITORY#*/}_notebook:${GITHUB_REF##*/}_${{ steps.date.outputs.date }} ./viz_scripts
```

Comment:
we should not be building the images using `docker build`
The build using GitHub actions and the README should be consistent.
The GitHub action should validate that the README and the README instructions are correct.
We should follow the instructions in the README and build using `docker-compose -f docker-compose.yml build`

Reply:
Okay, I changed this to use docker compose. In order to do so, I used an environment variable to set the names of the images being built with compose to the proper image tag name, so that the subsequent image push command will work. The desired tag name was previously passed into the build command, which is not possible with docker compose. 

I also renamed the variable `DOCKER_IMAGE_TAG` to `SERVER_IMAGE_TAG`, as I think this is a more meaningful variable name and is helpful for discerning all of the different tags.